shared/util_test.go

```diff
@@ -52,3 +63,56 @@ func TestProductChannelToLabel(t *testing.T) {
 	assert.Equal(t, ExperimentalLabel, ProductChannelToLabel("experimental"))
 	assert.Equal(t, "", ProductChannelToLabel("not-a-channel"))
 }
+
+func TestGetResultsURL_EmptyFile(t *testing.T) {
+	checkResult(
+		t,
+		Case{
+			TestRun{
+				ProductAtRevision: ProductAtRevision{
+					Revision: shortSHA,
+				},
+				ResultsURL: resultsURL,
+			},
+			"",
+			resultsURL,
+		})
+}
+
+func TestGetResultsURL_TestFile(t *testing.T) {
+	file := "css/vendor-imports/mozilla/mozilla-central-reftests/flexbox/flexbox-root-node-001b.html"
+	checkResult(
+		t,
+		Case{
+			TestRun{
+				ProductAtRevision: ProductAtRevision{
+					Revision: shortSHA,
+				},
+				ResultsURL: resultsURL,
+			},
+			file,
+			resultsURLBase + product + "/" + file,
+		})
+}
+
+func TestGetResultsURL_TrailingSlash(t *testing.T) {
+	checkResult(
+		t,
+		Case{
+			TestRun{
+				ProductAtRevision: ProductAtRevision{
+					Revision: shortSHA,
+				},
+				ResultsURL: resultsURL,
+			},
+			"/",
+			resultsURL,
+		})
+}
+
+func checkResult(t *testing.T, c Case) {
```

Comment:
I'd drop `Case` and just pass the 3 params to this function. Will shave 10 or more lines of code.